src/transformers/pipelines/__init__.py

```diff
@@ -398,6 +410,7 @@
 NO_FEATURE_EXTRACTOR_TASKS = set()
 NO_IMAGE_PROCESSOR_TASKS = set()
 NO_TOKENIZER_TASKS = set()
+NO_PROCESSOR_TASKS = set()
```

Comment:
Pipelines are never using `processor`. And it cannot.
Processor are handy tools, but they just do too much and have no stable API, you can send text, images, audio and whatnot, with no possibility for the caller to know what is the correct contract (and inspecting the signature is kind of a sin).

`FeatureExtractor` and `Tokenizer` should be enough I think.

Reply:
> `FeatureExtractor` and `Tokenizer` should be enough I think

The `Tokenizer` is required for all models to pre-process the text inputs to input ids. The `FeatureExtractor` allows us to convert from log-mel spectrogram outputs to a waveform audio, so is only used for models that predict a log-mel outputs (i.e. SpeechT5 and FastSpeech2, not Bark)

Neither prepare the official speaker embeddings, which is only done with the `processor` (as discussed in the [PR](https://github.com/huggingface/transformers/pull/24086#discussion_r1248174185)). 

Since all future TTS models should have a `processor` with a consistent signature, it is surely ok to have a new pipeline class use the `processor`?